File: Longest Consecutive Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int> &num) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        unordered_map<int,bool> hmap;
        for(int &val:num)
        {
            if(hmap.count(val)==0)
            {
                hmap[val] = false;
            }
        }
        
        
        int N = num.size();
        int max = 0;
       
        for(int i = 0; i < N; i++)
        {
           
            int val = num[i];
            if(hmap[val]==false)
            {
                int count = 1;
                int up = val+1;
                while(hmap.count(up))
                {
                    hmap[up] = true;
                    count++;
                    up++;
                }
                
                int down = val-1;
                while(hmap.count(down))
                {
                    hmap[down] = true;
                    count++;
                    down--;
                }
                
                max = count>max?count:max; 
            }
        }
        
        return max;
    }
};

File: Longest Consecutive Sequence.java
public class Solution {
    public int longestConsecutive(int[] num) {
        // Start typing your Java solution below
        // DO NOT write main() function

        HashMap<Integer, Integer> hashmap = new HashMap<Integer, Integer>();
        
        int maxCount = 1;
        for(int i: num)
        {
            if(hashmap.containsKey(i))
            {
                continue;
            }
            
            hashmap.put(i, 1);
            
            if(hashmap.containsKey(i+1))
            {
                maxCount = Math.max(maxCount, getLen(hashmap, i + 1, i));
            }
            
            if(hashmap.containsKey(i-1))
            {
                maxCount = Math.max(maxCount, getLen(hashmap, i, i - 1));
            }

        }
        
        return maxCount;
    }
    
    private int getLen(HashMap<Integer, Integer> hashmap, int up, int low)
    {
        int upper = up + hashmap.get(up) - 1;
        int lower = low - hashmap.get(low) + 1;
        int len = upper - lower + 1;
        hashmap.put(upper, len);
        hashmap.put(lower, len);
        return len;
    }
    
}